flytekit/core/python_customized_container_task.py

```diff
@@ -105,7 +105,7 @@ def __init__(
         self._resources = ResourceSpec(
             requests=requests if requests else Resources(), limits=limits if limits else Resources()
         )
-        self._environment = environment
+        self._environment = environment  # type: ignore
```

Comment:
It's better to make these two variables become the same type instead of using ignore here.
```suggestion
        self._environment = environment or {}
```